cmake/detect-features-c.cmake

```diff
@@ -367,9 +367,43 @@ int main() {
 }
 " CMK_BALANCED_INJECTION_API)
 
-if(${CMK_BUILD_OFI} EQUAL 1)
+if(${NETWORK} STREQUAL "ofi" OR ${NETWORK} STREQUAL "ofi-crayshasta" OR ${NETWORK} STREQUAL "ofi-linux")
+# assume HPC installation
+include(CMakePrintHelpers)
+  find_package(EnvModules REQUIRED)
+  find_package(PkgConfig REQUIRED)
+  if(EnvModules_FOUND)
+  #at least get libfabric loaded if it isn't already
+	env_module(load libfabric)
+  endif()
```

Comment:
I think this assumption is too restrictive. It does not seem to work when building Charm++ with Spack, since `libfabric` is provided as a Spack package and not through modules.

Reply:
It makes sense for the standard case of building on a DOE machine, but could present a problem for Spack.  Is there is a simple way to test for the Spack case and then extract the necessary information to accomplish the same smooth build this code accomplishes with LMOD and PkgConfig?

Reply:
I think the simplest way is to setup Charm++ as a develop build and introduce changes to the build system and Spack package until it works. I started working on it but got stuck and then other stuff took priority.